JamGame/Assets/Scripts/Overlay/OverlayManager.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+public interface IOverlayManager
+{
+    public void RevertAllOverlays();
+
+    public void ApplyOverlay<O>(O overlay) where O : class, IOverlay;
+}
+
+public class OverlayManager : MonoBehaviour, IOverlayManager
```

Comment:
Really? Manager?

Reply:
any suggestions?

Reply:
Maybe switcher or Applier, if no - leave it as it is